sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/externalreport/TslintReportSensor.java

```diff
@@ -0,0 +1,232 @@
+    "no-construct",
+    "no-debugger",
+    "no-duplicate-super",
+    "no-duplicate-switch-case",
+    "no-duplicate-variable",
+    "no-dynamic-delete",
+    "no-empty",
+    "no-eval",
+    "no-floating-promises",
+    "no-for-in-array",
+    "no-implicit-dependencies",
+    "no-inferred-empty-object-type",
+    "no-invalid-template-strings",
+    "no-invalid-this",
+    "no-misused-new",
+    "no-null-keyword",
+    "no-object-literal-type-assertion",
+    "no-return-await",
+    "no-shadowed-variable",
+    "no-sparse-arrays",
+    "no-string-literal",
+    "no-string-throw",
+    "no-submodule-imports",
+    "no-switch-case-fall-through",
+    "no-this-assignment",
+    "no-unbound-method",
+    "no-unnecessary-class",
+    "no-unsafe-any",
+    "no-unsafe-finally",
+    "no-unused-expression",
+    "no-unused-variable",
+    "no-use-before-declare",
+    "no-var-keyword",
+    "no-void-expression",
+    "prefer-conditional-expression",
+    "prefer-object-spread",
+    "radix",
+    "restrict-plus-operands",
+    "strict-boolean-expressions",
+    "strict-type-predicates",
+    "switch-default",
+    "triple-equals",
+    "typeof-compare",
+    "use-default-type-parameter",
+    "use-isnan"));
+
+  public TslintReportSensor(CheckFactory checkFactory) {
+    TypeScriptRules typeScriptRules = new TypeScriptRules(checkFactory);
+    typeScriptRules.forEach(typeScriptRule -> {
+      if (typeScriptRule.isEnabled()) {
+        activatedTslintKeys.add(typeScriptRule.tsLintKey());
+      }
+    });
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    boolean externalIssuesSupported = context.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(7, 2));
+    String[] tslintReportPaths = context.config().getStringArray(TSLINT_REPORT_PATHS);
+
+    if (tslintReportPaths.length == 0) {
+      return;
+    } else if (!externalIssuesSupported) {
+      LOG.error("Current version of SonarQube doesn't support import of external issues (at least 7.2 required).");
+      return;
+    }
+
+    for (String reportPath : tslintReportPaths) {
+      File tslintReport = getIOFile(context.fileSystem().baseDir(), reportPath);
+      importTslintReport(tslintReport, context);
+    }
+  }
+
+  private void importTslintReport(File report, SensorContext context) {
+    LOG.info("Importing {}", report.getAbsoluteFile());
+
+    try (InputStreamReader inputStreamReader = new InputStreamReader(new FileInputStream(report), StandardCharsets.UTF_8)) {
+      TslintError[] tslintErrors = gson.fromJson(inputStreamReader, TslintError[].class);
+      for (TslintError tslintError : tslintErrors) {
+        saveTslintError(context, tslintError);
+      }
+    } catch (IOException e) {
+      LOG.error("No TSLint issues information will be saved because file cannot be found.", e);
+    }
+  }
+
+  private void saveTslintError(SensorContext context, TslintError tslintError) {
+    if (activatedTslintKeys.contains(tslintError.ruleName)) {
+      return;
```

Comment:
Maybe we can have `DEBUG` level log here, so user can investigate why issue is not imported, something like "Issue on line 42 skipped because the same issue will be reported by SonarTS"

Reply:
@saberduck great idea!